browser/views.py

```diff
@@ -687,3 +687,109 @@ def delete_investigator_record(request, person_id, research_id):
         Investigator.objects.filter(id=research_id).delete()
 
     return HttpResponseRedirect(reverse('person', args=(person_id,)))
+
+
+@staff_member_required
+def add_position(request, person_id):
+    person = get_object_or_404(Person, pk=person_id)
+    template = "browser/position.html"
+    form = PositionForm()
+
+    if request.method == 'POST':
+        form = PositionForm(request.POST, instance=person)
+        if form.is_valid():
+            # For optional start and end dates we need to set the result as None, otherwise a database error is thrown
+            # Initially they are set as null, and set according to user data if a date has been set
+            start_date = None
+            end_date = None
+            if '/' in request.POST.get("start_date"):
+                # Datepicker is returning the format as mm/dd/yyyy so, converting it into yyyy-mm-dd
+                start_date = request.POST.get("start_date")
+                start_month = start_date.split('/')[0]
+                start_year = start_date.split('/')[2]
+                start_day = start_date.split('/')[1]
+                start_date = start_year + '-' + start_month + '-' + start_day
+
+            if '/' in request.POST.get("end_date") :
+                # Datepicker is returning the format as mm/dd/yyyy so, converting it into yyyy-mm-dd
+                end_date = request.POST.get("end_date")
+                end_month = end_date.split('/')[0]
+                end_year = end_date.split('/')[2]
+                end_day = end_date.split('/')[1]
+                end_date = end_year + '-' + end_month + '-' + end_day
```

Comment:
if you use a datefield in the form for those fields, then it should already be validated and converted, correctly, shouldn't it? This seems very hacky.

Reply:
Removed this date conversion python code and instead, changed the date format of the html date picker. Date picker returns the correct format which need not be converted again in the python side.

Reply:
how about validation? shouldn't the datefield do that for you? what kind of field is it right now? String or Date?

Reply:
The field being returned to back-end is of string data type